File: apply-migrations.ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

async function applyMigrations() {
  try {
    console.log('🚀 Aplicando migraciones necesarias...\n');

    // 1. Marcar admin como Owner
    console.log('1️⃣  Marcando a CarlosAdmin como Owner...');
    const updatedUser = await prisma.users.updateMany({
      where: { username: 'CarlosAdmin' },
      data: { is_owner: true }
    });
    console.log(`   ✅ ${updatedUser.count} usuario(s) actualizado(s)\n`);

    // 2. Verificar que se actualizó
    const adminUser = await prisma.users.findFirst({
      where: { username: 'CarlosAdmin' },
      select: { username: true, email: true, is_owner: true }
    });
    if (adminUser) {
      console.log('   📋 Verificación:');
      console.log(`      Usuario: ${adminUser.username}`);
      console.log(`      Email: ${adminUser.email}`);
      console.log(`      Is Owner: ${adminUser.is_owner ? '✅ true' : '❌ false'}\n`);
    }

    // 3. Obtener información de roles
    console.log('2️⃣  Verificando tabla roles...');
    const roles = await prisma.roles.findMany({
      select: {
        id: true,
        name: true,
        is_system: true
      }
    });
    
    console.log(`   ✅ Total de roles: ${roles.length}`);
    roles.forEach((role: any) => {
      console.log(`      - ${role.name}${role.is_system ? ' (SISTEMA)' : ''}`);
    });

    console.log('\n✅ ¡Migraciones aplicadas exitosamente!');
    console.log('\n📋 Resumen:');
    console.log('   ✅ CarlosAdmin marcado como Owner');
    console.log('   ✅ Tabla roles lista para nuevos campos (color, icon, permisos, etc)');
    console.log('\n🚀 Próximo paso: Reinicia el servidor y verifica en el endpoint /api/auth/profile');

  } catch (error) {
    console.error('❌ Error aplicando migraciones:', error);
    process.exit(1);
  } finally {
    await prisma.$disconnect();
  }
}

applyMigrations();

File: apply-migrations.py
#!/usr/bin/env python3

import requests
import json
import sys

BASE_URL = "http://localhost:5001"

# 1. Login
print("🔐 Obteniendo token...")
login_response = requests.post(
    f"{BASE_URL}/api/auth/login",
    json={
        "username": "CarlosAdmin",
        "password": "Turleque2026$"
    }
)

if login_response.status_code != 200:
    print(f"❌ Error en login: {login_response.status_code}")
    print(login_response.text)
    sys.exit(1)

login_data = login_response.json()
token = login_data.get('token')
user_id = login_data.get('id')

if not token:
    print("❌ No se obtuvo token")
    sys.exit(1)

print(f"✅ Token obtenido: {token[:50]}...")
print(f"👤 User ID: {user_id}")
print()

# 2. Aplicar migraciones
print("🚀 Aplicando migraciones...")
migration_response = requests.post(
    f"{BASE_URL}/api/admin/apply-migrations",
    headers={"Authorization": f"Bearer {token}"},
    json={}
)

if migration_response.status_code != 200:
    print(f"❌ Error aplicando migraciones: {migration_response.status_code}")
    print(migration_response.text)
    sys.exit(1)

migration_data = migration_response.json()
print(json.dumps(migration_data, indent=2))

# 3. Verificar que el admin tiene is_owner = true
print()
print("✅ Verificando admin...")
profile_response = requests.get(
    f"{BASE_URL}/api/auth/profile",
    headers={"Authorization": f"Bearer {token}"}
)

profile_data = profile_response.json()
is_owner = profile_data.get('is_owner')

print(f"Is Owner: {is_owner}")

if is_owner:
    print()
    print("✅ ¡MIGRACIÓN COMPLETADA EXITOSAMENTE!")
    print("   - Admin marcado como Owner: ✅")
    print("   - Roles prontos para nuevos campos: ✅")
    print()
    print("📋 Próximo paso: Reinicia el servidor para que cargue los cambios")
else:
    print()
    print("⚠️  El admin no fue marcado como Owner")

